src/ResourceManager/ManagedServiceIdentity/Commands.ManagedServiceIdentity/UserAssignedIdentities/GetAzureRmUserAssignedIdentityCmdlet.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ManagedServiceIdentity.Common;
+using Microsoft.Azure.Commands.ManagedServiceIdentity.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.ManagedServiceIdentity.UserAssignedIdentities
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmUserAssignedIdentity")]
+    [OutputType(typeof (PsUserAssignedIdentity))]
+    public class GetAzureRmUserAssignedIdentityCmdlet : MsiBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            Position = 0,
+            HelpMessage = "The resource group name.")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 1,
+            HelpMessage = "The Identity name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+
+            ExecuteClientAction(() =>
+            {
+                if (string.IsNullOrEmpty(this.ResourceGroupName))
```

Comment:
Question -
Get-AzureRmUserAssignedIdentity can be called in 3 ways:
Get-AzureRmUserAssignedIdentity -ResourcegroupName XXX
Get-AzureRmUserAssignedIdentity -ResourceGroupName XXX -Name YYY

I have specified these two parameter sets:

[Parameter(ParameterSetName="SubscriptionOrRGLevel", Mandatory="False")]
[Parameter(ParameterSetName="RGLOrIdentityLevel", Mandatory="True")]
public string ResourceGroupName

[Parameter(ParameterSetName="RGLOrIdentityLevel", Mandatory="False")]
public string IdentityName

Now, since SubscriptionOrRGLevel does not have any unique parameter, request of type: Get-AzureRmUserAssignedIdentity -ResourceGroupName XXX fail for not being able to identify parameter set. Given this, how should I go about implementing the two suggested parameter sets